src/clj/game/core/abilities.clj

```diff
@@ -278,20 +278,30 @@
 
 ;;; Optional Ability
 (defn optional-ability
-  "Shows a 'Yes/No' prompt and resolves the given ability if Yes is chosen."
+  "Shows a 'Yes/No' prompt and resolves the given ability's :yes-ability if Yes is chosen, and :no-ability otherwise.
+  If ability has an :autoresolve entry, first call it as a 5-function, and if it returns 'Yes' or 'No'
+  resolve the ability as if prompt was displayed and Yes/No was chosen."
   ([state side card message ability targets] (optional-ability state side (make-eid state) card message ability targets))
   ([state side eid card message ability targets]
-   (show-prompt state side eid card message ["Yes" "No"]
-                #(let [yes-ability (:yes-ability ability)]
-                  (if (and (= % "Yes")
-                           yes-ability
-                           (can-pay? state side (:title card) (:cost yes-ability)))
-                    (resolve-ability state side (assoc yes-ability :eid eid) card targets)
-                    (if-let [no-ability (:no-ability ability)]
-                      (resolve-ability state side (assoc no-ability :eid eid) card targets)
-                      (effect-completed state side eid))))
-                ability)))
-
+   (letfn [(prompt-fn [prompt-choice]
+             (let [yes-ability (:yes-ability ability)]
+               (if (and (= prompt-choice "Yes")
+                        yes-ability
+                        (can-pay? state side (:title card) (:cost yes-ability)))
+                 (resolve-ability state side (assoc yes-ability :eid eid) card targets)
+                 (if-let [no-ability (:no-ability ability)]
+                   (resolve-ability state side (assoc no-ability :eid eid) card targets)
+                   (effect-completed state side eid)))))]
+     (let [autoresolve-as (when-let [autoresolve-fn (:autoresolve ability)]
+                            (autoresolve-fn state side eid card targets))]
+       (case (when-let [autoresolve-fn (:autoresolve ability)]
```

Comment:
This is merely a readability/style thing, but would you mind changing this to a `when-let*` and assigning `(autoresolve-fn state side eid card targets)` to a variable, and then saying `case variable-name`? It's slightly wordier, but it helps understanding this line a lot.

Reply:
Yeah, I'll do that. To be honest, I don't actually understand this line when I reread it. Why am I binding the name `autoresolve-as` without using it? So some cleanup is definitely in order.